control-plane/csi-driver/src/context.rs

```diff
@@ -414,6 +425,33 @@ impl TryFrom<&HashMap<String, String>> for PublishParams {
     }
 }
 
+/// Validate the topology parameters.
+pub(crate) fn validate_single_topology_param(
+    pool_affinity_topology_label: &Option<HashMap<String, String>>,
+    pool_affinity_topology_key: &Option<Vec<String>>,
+    pool_has_topology_key: &Option<HashMap<String, String>>,
+    node_affinity_topology_label: &Option<HashMap<String, String>>,
+    node_affinity_topology_key: &Option<Vec<String>>,
+    node_has_topology_key: &Option<HashMap<String, String>>,
+    node_spread_topology_key: &Option<HashMap<String, String>>,
+) -> Result<(), tonic::Status> {
+    let mut number_of_topology_params = 0;
+    number_of_topology_params += pool_affinity_topology_label.is_some() as usize;
+    number_of_topology_params += pool_affinity_topology_key.is_some() as usize;
+    number_of_topology_params += pool_has_topology_key.is_some() as usize;
+    number_of_topology_params += node_affinity_topology_label.is_some() as usize;
+    number_of_topology_params += node_affinity_topology_key.is_some() as usize;
+    number_of_topology_params += node_has_topology_key.is_some() as usize;
+    number_of_topology_params += node_spread_topology_key.is_some() as usize;
+
+    if number_of_topology_params > 1 {
+        return Err(tonic::Status::invalid_argument(
+            "Multiple topology parameters are not allowed",
+        ));
+    }
+    Ok(())
```

Comment:
```suggestion
    let topology_params = [
        pool_affinity_topology_label,
        pool_affinity_topology_key,
        pool_has_topology_key,
        node_affinity_topology_label,
        node_affinity_topology_key,
        node_has_topology_key,
        node_spread_topology_key,
    ];

    if topology_params.iter().any(|param| param.is_some()) {
        return Err(tonic::Status::invalid_argument(
            "Topology parameters are not allowed",
        ));
    }
   Ok(())
```
Can we just do this? @tiagolobocastro wdyt?

Reply:
The counting solution is also fine. @sinhaashish upto you.

Reply:
Ohh no, my bad, we need only one, my solution is incorrect in that case.

```suggestion
   let topology_params = [
        pool_affinity_topology_label,
        pool_affinity_topology_key,
        pool_has_topology_key,
        node_affinity_topology_label,
        node_affinity_topology_key,
        node_has_topology_key,
        node_spread_topology_key,
    ];

    let number_of_topology_params = topology_params.iter().filter(|&&param| param.is_some()).count();

    if number_of_topology_params > 0 {
        return Err(tonic::Status::invalid_argument(
            "Topology parameters are not allowed",
        ));
    }
    Ok(())
```

How about this one?

Reply:
Out of all these params the affinity keys are vec of strings

so the above code errros out. 
pool_affinity_topology_key,
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `&Option<HashMap<String, String>>`, found `&Option<Vec<String>>`

Reply:
Updated it , iter() for hashmap and increement for vec of strings

Reply:
You can use is_some() in the list elements to make the list a list of bools, and then check count for true.

Reply:
```
 let topology_params = [
      pool_affinity_topology_label.is_some(),
      pool_affinity_topology_key.is_some(),
      pool_has_topology_key.is_some(),
      node_affinity_topology_label.is_some(),
      node_affinity_topology_key.is_some(),
      node_has_topology_key.is_some(),
      node_spread_topology_key.is_some(),
  ];

  let number_of_topology_params = topology_params.iter().filter(|&&param| param).count();

  if number_of_topology_params > 1 {
      return Err(tonic::Status::invalid_argument(
          "Multiple topology parameters are not allowed",
      ));
  }

  Ok(())
```

Reply:
Something like this?

Reply:
done